packages/expo-intent-launcher/src/IntentLauncher.ts

```diff
@@ -101,15 +101,32 @@ export const ACTION_ZEN_MODE_SCHEDULE_RULE_SETTINGS =
   'android.settings.ZEN_MODE_SCHEDULE_RULE_SETTINGS';
 export const ACTION_ZEN_MODE_SETTINGS = 'android.settings.ZEN_MODE_SETTINGS';
 
-export function startActivityAsync(
-  activity: string,
-  // TODO: Make this type more precise
-  data: { [key: string]: any } | null = null,
-  uri: string | null = null,
-  mime: string | null = null
-): Promise<boolean> {
-  if (!ExponentIntentLauncher.startActivity) {
-    throw new UnavailabilityError('IntentLauncherAndroid', 'startActivityAsync');
+interface IntentParams {
+  action?: string;
+  type?: string;
+  category?: string;
+  extra?: object;
+  data?: string;
+  flags?: number;
+  packageName?: string;
+  className?: string;
+}
+
+interface IntentResult {
+  resultCode: number;
+  data?: string;
+  extra?: object;
+}
+
+export enum ResultCode {
+  Success = -1,
+  Canceled = 0,
+  FirstUser = 1,
+}
+
+export async function startActivityAsync(params: IntentParams = {}): Promise<IntentResult> {
```

Comment:
I'm not sure I'm really a fan of putting all the args into a big object like this, I would rather leave `activity:string` as the first, required argument and then let developers pass extra `data, uri, mime` in an options object as the second argument, but that's only my opinion 🤷‍♂️

Reply:
@sjchmiela but I cannot see `activity` argument anymore. All these params seems to be optional now, right @tsapeta?

Reply:
`activity` has been replaced with `action` and actually there is a mistake in the interface above, `action` argument is required.

Reply:
Ah, so I would agree with @sjchmiela on having this argument as separate parameter 😄

Reply:
Summon @dsokal 	😛